sklearn/linear_model/least_angle.py

```diff
@@ -680,24 +687,27 @@ def fit(self, X, y, Xy=None):
                                    self.coef_)]
                 self.n_iter_ = self.n_iter_[0]
         else:
+            lars_paths = Parallel(n_jobs=self.n_jobs, verbose=self.verbose)(
+                delayed(lars_path)(
+                    X, y[:, k], Gram=Gram, Xy=None if Xy is None else Xy[:, k],
```

Comment:
Please correct if i am wrong but I think that would involve having a new function which will process each slice and make calls to lars_path separately for each target column. I am not sure if this would lead to significant speed gains.